cpp/tests/hashing/sha1_test.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/hashing.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+constexpr cudf::test::debug_output_level verbosity{cudf::test::debug_output_level::ALL_ERRORS};
+
+class SHA1HashTest : public cudf::test::BaseFixture {};
+
+TEST_F(SHA1HashTest, EmptyTable)
+{
+  auto const empty_table        = cudf::table_view{};
+  auto const empty_column       = cudf::make_empty_column(cudf::data_type(cudf::type_id::STRING));
+  auto const output_empty_table = cudf::hashing::sha1(empty_table);
+  EXPECT_EQ(empty_column->size(), output_empty_table->size());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(empty_column->view(), output_empty_table->view());
+
+  auto const table_one_empty_column  = cudf::table_view{{empty_column->view()}};
+  auto const output_one_empty_column = cudf::hashing::sha1(empty_table);
+  EXPECT_EQ(empty_column->size(), output_one_empty_column->size());
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(empty_column->view(), output_one_empty_column->view());
+}
+
+TEST_F(SHA1HashTest, MultiValue)
+{
+  cudf::test::strings_column_wrapper const strings_col(
+    {"",
+     "0",
+     "A 56 character string to test message padding algorithm.",
+     "A 63 character string to test message padding algorithm, again.",
+     "A 64 character string to test message padding algorithm, again!!",
+     "A very long (greater than 128 bytes/char string) to execute a multi hash-step data point in "
+     "the hash function being tested. This string needed to be longer.",
+     "All work and no play makes Jack a dull boy",
+     "!\"#$%&\'()*+,-./0123456789:;<=>?@[\\]^_`{|}~"});
```

Comment:
I would recommend also including some multi-byte characters. Here are a few simple ones `é¼³⅝`
Just curious how were the results computed/verified for these?

Reply:
The results can be verified with Python's `hashlib`.

```python
import hashlib

# Input string
input_string = input("Enter a string: ")

# Create a SHA-256 hash object
hash_object = hashlib.sha256()

# Update the hash object with the input string
hash_object.update(input_string.encode())

# Get the hexadecimal representation of the hash
hex_digest = hash_object.hexdigest()

# Print the SHA-256 hash
print("SHA-256 Hash:", hex_digest)
```

I also checked a third implementation with `sha256sum` (note that you have to be careful with line/string termination characters if doing this in a shell, or you'll get different results).
```bash
$ echo -n "" | sha256sum
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

Reply:
I think it would be good to include this information in the comments here perhaps.
In case additional test values are added in the future.

Reply:
This is done in 186eace705f00b815c2a8962a8a92fbdda922adf.